internal/business/server/http_server.go

```diff
@@ -0,0 +1,80 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+
+	"github.com/samber/oops"
+
+	slogctx "github.com/veqryn/slog-context"
+
+	"github.com/openkcm/session-manager/internal/config"
+	"github.com/openkcm/session-manager/internal/openapi"
+	"github.com/openkcm/session-manager/internal/session"
+)
+
+// createStatusServer creates an API http server using the given config
+func createHTTPServer(_ context.Context, cfg *config.Config, sManager *session.Manager) *http.Server {
+	mux := http.NewServeMux()
+	openAPIServer := newOpenAPIServer(sManager)
+	strictHandler := openapi.NewStrictHandler(
+		openAPIServer,
+		[]openapi.StrictMiddlewareFunc{
+			newTraceMiddleware(cfg),
+		},
+	)
+
+	smHandler := openapi.Handler(strictHandler)
+	mux.Handle("/sm", smHandler)
+
+	return &http.Server{
+		Addr:    cfg.HTTP.Address,
+		Handler: mux,
+	}
+}
+
+// StartHTTPServer starts the gRPC server using the given config.
+func StartHTTPServer(ctx context.Context, cfg *config.Config, sManager *session.Manager) error {
+	if err := initMeters(ctx, cfg); err != nil {
+		return err
+	}
+
+	server := createHTTPServer(ctx, cfg, sManager)
+
+	slogctx.Info(ctx, "Starting HTTP listener", "address", server.Addr)
+
+	listener, err := new(net.ListenConfig).Listen(ctx, "tcp", server.Addr)
```

Comment:
```suggestion
	listener, err := net.Listen(ctx, "tcp", server.Addr)
```
cannot see where you fill "net.ListenConfig" so we can skip it?